common/rdm/ResponderPersonality.cpp

```diff
@@ -40,6 +40,10 @@ using std::string;
 Personality::Personality(uint16_t footprint, const string &description)
     : m_footprint(footprint),
       m_description(description) {
+        STATIC_ASSERT(sizeof(description) <= 32);
+        // static_assert(
+        //   sizeof(description) <= 32,
+        //   "Personality Description is to long (must be <= 32 chars)");
 }
 
 Personality::Personality(uint16_t footprint, const string &description,
```

Comment:
ok i missed the second constructor.
but also with the same line adde there it does not throw an error for the 'to long' description..
so it seems iam missing something else...

Reply:
You want description.length() to find out how many characters are in the string, sizeof tells you the size of the object (which I think will be a pointer in this case).

Reply:
Does "STATIC_ASSERT(description.length() <= 32);" not work then? Are you importing Macro.h for our STATIC_ASSERT macro?

Reply:
Does "STATIC_ASSERT(description.length() <= 32);" not work then? Are you importing Macro.h for our STATIC_ASSERT macro?

Reply:
i did not `#include "ola/base/Macro.h"` but it seems it is included (this was not the problem except it is overwritting some other defaults) the error message if i uncomment this is:
```
common/rdm/ResponderPersonality.cpp:44:41: error: call to non-constexpr function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::length() const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
         STATIC_ASSERT(description.length() <= 32);
                       ~~~~~~~~~~~~~~~~~~^~
./include/ola/base/Macro.h:125:66: note: in definition of macro ‘STATIC_ASSERT’
     sizeof(::internal::STATIC_ASSERT_FAILURE< static_cast<bool>( x ) >)>\
```
so if i interpret this correctly it tells me that `.length()`does return a `non-constexpr` and that STATIC_ASSERT needs a `constexpr` --that makes sense as the code is not run because the STATIC_ASSERT is evaluated at compile time.. - as far [as i found](https://stackoverflow.com/questions/27123306/is-it-possible-to-use-stdstring-in-a-constexpr#27123850) there is no way to get a `std:string` to be constexpr compatible

Reply:
Yes apologies, I did some testing afterwards and found the same thing, but failed to comment.

I suspect, but haven't double checked, that OLA will already truncate the returned string to within the RDM spec (i.e. 32 characters). So perhaps we should just compromise on adding an OLA_WARN or OLA_CRIT to the constructors, so we at least see a message in the log if someone does something they shouldn't. I did find some stuff about string literals, which may possibly mean we could do it with a macro, which eventually passes the literal into the std::string, but I'm not certain about that.